src/data_api/events.se

```diff
@@ -1,346 +1,420 @@
+def assertEventIsInDisputePhase(eventId):
+    self.assertEventIsInCurrentReportingWindow(eventId)
+    reportingContract = self.controller.lookup('reporting')
+    if (!reportingContract.isInReportingDisputePhase()):
+        ~invalid()
     return(1)
 
-# @return fxp
-def getUncaughtOutcome(event):
-    self.controller.assertIsWhitelisted(msg.sender)
-    return(self.Events[event].uncaughtOutcome)
-
-def initializeEvent(ID, branch, expirationDate, fxpMinValue, fxpMaxValue, numOutcomes, resolution: str, resolutionAddress, resolveBondPoster, forkResolveAddress):
-    self.controller.assertIsWhitelisted(msg.sender)
-    if(!self.Events[ID].expirationDate):
-        self.Events[ID].branch = branch
-        self.Events[ID].expirationDate = expirationDate
-        self.Events[ID].originalExp = expirationDate
-        self.Events[ID].fxpMinValue = fxpMinValue
-        self.Events[ID].fxpMaxValue = fxpMaxValue
-        self.Events[ID].numOutcomes = numOutcomes
-        self.Events[ID].resolveBondPoster = resolveBondPoster
-        resolutionSourceLength = len(resolution)
-        self.Events[ID].resolutionSourceLength = resolutionSourceLength
-        save(self.Events[ID].resolutionSource[0], resolution, chars=resolutionSourceLength)
-        self.Events[ID].creationTime = block.timestamp
-        self.Events[ID].resolutionAddress = resolutionAddress
-        self.Events[ID].forkResolveAddress = forkResolveAddress
-        return(1)
-    else:
-        return(0)
-
-def setBond(event, bond):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].bond = bond
+def assertEventIsNotFinalized(eventId):
+    if (self.isEventFinalized(eventId)):
+        ~invalid()
     return(1)
 
-def getEarlyResolutionBond(event):
-    return(self.Events[event].earlyResolutionBond)
-
-def setEarlyResolutionBond(event, bond):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].earlyResolutionBond = bond
+def assertEventIsNotDoneWithReporters(eventId):
+    if (self.events[eventId].allReportersOutcome != 0):
+        ~invalid()
     return(1)
 
-def addMarket(event, market):
-    self.controller.assertIsWhitelisted(msg.sender)
-    self.Events[event].markets[self.Events[event].numMarkets] = market
-    self.Events[event].numMarkets += 1
+def assertReporterHasRegistrationToken(reporterAddress):
+    registrationTokenContract = self.controller.lookup('registrationToken')
+    branch = self.events[eventId].branch
+    reportingWindow = self.events[eventId].reportingWindow
+    numberOfRegistrationTokens = registrationTokenContract.registrationTokenBalance(branch, reportingWindow)
+    if (numberOfRegistrationTokens < 1):
+        ~invalid()
     return(1)
 
-def getEventBranch(event):
-    return(self.Events[event].branch)
+# TODO: implement ERC20-like interface for each of the 3 different bond tokens
+# CONSIDER: is it worth making them ERC20-like?  Perhaps just add a transfer/balance function instead?
```

Comment:
They ended up as ERC20 tokens in @adrake33's changes mainly because they ended up as separate contracts to simplify code and help get some code out of market.se (which is too big) and adding the ERC20 interface was trivial at that point (just copy/paste).